### ISSUE TITLE
real and imag raise exceptions on non-complex numbers

### ISSUE THREAD

**Issue by HugoNumworks** [micropython/micropython#6635]
The real or imaginary part of a non-complex number cannot be computed using MicroPython, whereas Python and CPython handle it.
Should I make a PR for this ?
It would be handled with MICROPY_PY_BUILTINS_COMPLEX only.
<img width="603" alt="image" src="https://user-images.githubusercontent.com/67104016/99661222-d0e92000-2a63-11eb-9157-7768d2169e67.png">

**Comment by dlech** [comment]
What is the use case for this?

Is `complex(2).real` a suitable workaround?

**Comment by peterhinch** [comment]
I guess the use case is where at runtime a variable might contain a real or a complex.

**Comment by HugoNumworks** [comment]
Indeed, but it would have been preferable not having to consider the number's type, like with Python. Same might go with conjugate() method.
